src/verinfast/dependencies/walkers/composer.py

```diff
@@ -0,0 +1,70 @@
+import json
+import os
+import subprocess
+from pathlib import Path
+from typing import List
+
+from verinfast.dependencies.walkers.classes import Walker, Entry
+
+
+class ComposerWalker(Walker):
+    dependency_tree = {}
+
+    def initialize(self, root_path: str = "./"):
+        root_path = str(Path(root_path).absolute())
+        self.install_points: List[Path] = []
+        for p in Path(root_path).rglob('**/*.*'):
+            if p.name in self.manifest_files:
+                self.install_points.append(p)
+        for p in self.install_points:
+            target_dir = Path(p).parent
+            os.chdir(target_dir)
+            try:
+                subprocess.run(
+                    "composer install --no-dev --no-progress",
+                    capture_output=True,
+                    shell=True
+                )
+            except Exception as error:
+                self.log(f'Error with composer install: {error}')
+            else:
+                self.walk('./')
+            os.chdir(root_path)
+
+    def parse(self, file: str, expand=False):
+        try:
+            with open(file) as f:
+                cjson = json.load(f)
+                if (
+                    "license" in cjson and
+                    self.dependency_tree.get(cjson['name'])
+                ):
+                    for version in self.dependency_tree[cjson['name']]:
```

Comment:
Because we never know the package version through `composer.json` if we see a license for a package we add it for all versions. I wasn't sure how we'd want to handle it so this is my naive approach.